src/test/fuzz/netmsg.cpp

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2020-2023 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <banman.h>
+#include <blockencodings.h>
+#include <consensus/consensus.h>
+#include <protocol.h>
+#include <streams.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <test/util/mining.h>
+#include <test/util/setup_common.h>
+#include <test/util/validation.h>
+#include <util/check.h>
+#include <util/time.h>
+#include <validationinterface.h>
+
+#include <chrono>
+#include <string>
+#include <thread>
+
+namespace {
+NetTestingSetup* g_setup;
+} // namespace
+
+static size_t& GetNumMsgTypes()
+{
+    static size_t g_num_msg_types{0};
+    return g_num_msg_types;
+}
+#define FUZZ_TARGET_MSG(msg_type)                          \
+    struct msg_type##CountBeforeMain {                     \
+        msg_type##CountBeforeMain()                        \
+        {                                                  \
+            ++GetNumMsgTypes();                            \
+        }                                                  \
+    } const static g_##msg_type##count_before_main;        \
+    FUZZ_TARGET_INIT(netmsg_##msg_type, initialize_netmsg) \
+    {                                                      \
+        fuzz_target(buffer, #msg_type);                    \
+    }
```

Comment:
Why not de-duplicate by putting it in the existing `./src/test/fuzz/process_message.cpp` file?

Reply:
You mean to put the entire contents of the newly introduced `fuzz/netmsg.cpp` inside `fuzz/process_message.cpp`?

Reply:
yes

Reply:
Looks plausible. I will try to do that and see how it will look.

Reply:
Done. This also reduced the size of the diff (the last commit). Thanks!

Reply:
I put the new e2e tests in a separate new file because this code (which I was trying to avoid duplicating) is gone after https://github.com/bitcoin/bitcoin/pull/27766